### PR TITLE
chore: impl `Error` on near-primitives errors that missed them

### PR DESCRIPTION
**Pull request by ChaoticTempest**
Saw that we were missing a couple impls for `std::error::Error` for some of the errors in near-primitives. Would be useful if we could convert them into `&dyn Error` or `anyhow` errors to be used in libraries such as `workspaces`

### CODE REVIEW

File: core/primitives/src/errors.rs [hunk 1]
```diff
@@ -21,6 +21,8 @@ pub enum TxExecutionError {
     InvalidTxError(InvalidTxError),
 }
 
+impl std::error::Error for TxExecutionError {}
```
Comment (miraclx) [inline core/primitives/src/errors.rs:24]: To be clear, `TxExecutionError` isn't directly an `Rpc*Error` but is instead embedded deep inside `FinalExecutionOutcomeView` that's the `Rpc*Response` of both the `broadcast_tx_commit` and `tx` RPC methods.

**Review by miraclx — APPROVED** [review]
Nice catch. LgTm!

**Comment by ChaoticTempest** [comment]
> > Would be useful if we could convert them into &dyn Error or anyhow errors to be used in libraries such as workspaces
> 
> I'd recommend against converting near-primitives errors directly to `anyhow`: this makes `near-primitives` a public dependency, as the underlying error can be extracted via `downcasting`. This not suuper important though.

@matklad gotcha, this conversation can probably be moved over to workspaces, but it's just that some error handling needs specific error data on the user side if they want to handle it. The thing was that these error types coming from `near-primitives` are pretty sprawling and large in number, so maintaining a 1 to 1 might prove to be difficult if we were to just provide our own set of error types in workspaces. I can see mismatch errors happening when we use a newer version of sandbox that provides newer types. I'm already seeing similar issues with `near-jsonrpc-client` where it fails to deserialize some newer error types. What would you recommend in this case?

File: core/primitives/src/errors.rs (hunk 1 above)
Comment (matklad) [reply core/primitives/src/errors.rs:24]: Hm, taking a second look here, this impl in particular feels suspect to me. `TxExecutionError` should **not** be an `std::error::Error`, it should **not** work with the `?` syntax. 

It's not a runtime error a-la io error. It is a value which signifies that the tx was succesfully processed, and that, as a result of this processing, the runtime concluded that the tx is invalid.

**Pull request merged**